docs/namespaces.md

```diff
@@ -0,0 +1,67 @@
+- `/remote.php/dav/files/<userid>/`
+- `/remote.php/webdav/`
+
+## Paths
+
+Whatever follows the two endpoints is the *relative path* of a file in a users home storage. These two URLs actually point to the same file for the user with the id `u-u-i-d`:
+- `https://cloud.example.org/dav/files/<userid>/path/to/file.md`
+- `https://cloud.example.org/webdav/path/to/file.md`
+
+Since `path/to/file.md` is only relative to the users home storage we need to prefix it with a path to the users home storage in the CS3 namespace.
+
+## Service flow
+
+Before we can dig into the namespaces we should have a look at the services that 
+For WebDAV requests:
+
+{{< mermaid class="text-center">}}
+graph LR
+client-- WebDAV -->proxy
+proxy-- WebDAV  --> ocdav
+ocdav-- CS3 --> reva-gateway
+reva-gateway-- CS3 --> storageprovider
+{{< /mermaid >}}
+
+GET and PUT for WebDAV use the data providers instead of storage providers. For a resumable upload the [TUS protocol](https://tus.io/) is used.
+
+Up and downloads take a different route so they can be scaled independently:
+
+{{< mermaid class="text-center">}}
+graph LR
+client-- tus -->proxy
+proxy-- tus --> ocdav
+ocdav-- tus --> reva-datagateway
+reva-datagateway-- tus --> dataprovider
+{{< /mermaid >}}
+
+
+## Namespaces
+
+
+| scenario                                                                               | endpoint                                                                            | relative path               | relevant ocdav config                                               | cs3 namespace                                                        | storage_provider_config                                                                                                                                                                                                                                            | user layout example                                                     | internal path                                                                                                                               |
+|----------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------|-----------------------------|---------------------------------------------------------------------|----------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------|
```

Comment:
add bullets or stars in front of the config settings, to make them more readable